driver-core/src/main/com/mongodb/client/model/expressions/Expression.java

```diff
@@ -21,12 +21,68 @@
 import java.util.function.Function;
 
 /**
- * <p>Users should treat these interfaces as sealed, and must not implement any
- * sub-interfaces.
+ * A value in the context of the MongoDB Query Language (MQL).
  *
- * TODO-END: 'cause an error', 'execution context', wrong types/unsafe operations
- * TODO-END: types and how missing/null are not part of any type.
- * TODO-END: behaviour of equals
+ * <p>The API provided by this base type and its subtypes is the Java-native
+ * variant of MQL. It is used to query the MongoDB server, to perform remote
+ * computations, to store and retrieve data, or to otherwise work with data
+ * on a MongoDB server or compatible execution context.
+ *
+ * <p>The following is an example of usage within an aggregation pipeline. Here,
+ * the current document value is obtained and its "numberArray" field is
+ * filtered and summed, in a style similar to that of the Java Stream API:
+ *
+ * <pre>{@code
+ * AggregateIterable<Document> result = col.aggregate(Arrays.asList(
+ *     addFields(new Field<>("result", current()
```

Comment:
I have added `import static com.mongodb.client.model.expressions.Expressions.current;` - I worry that people will think it is a convention to write `MqlValue.current()`, rather than just something we do to make the imports clear.

For your second point, I think because this method is on the "border" of Java and the existing agg, rather than strictly part of the API, that it should be "contained" into this class, which deals with this "border".